### ISSUE TITLE
NOMINATE Plot: Porting to low resolution context, margins, and other display issues

### ISSUE THREAD

**Issue by aaronrudkin** [JeffreyBLewis/WebVoteView#106]
I wired up the district page to be able to display NOMINATE results. There are at least two or three major issues with it, though:

1. The axis and margins are quite space intensive and mess up totally if I try to make the plot small
2. If I modify the exterior margins, the points don't plot properly on the oval overlay--even changing the dimensions of the plot seems to mess up the overlay, which makes me think the large overlay on the Congress page is actually wrong.
3. Even if it displayed properly. it's hard to parse meaning from it without a sense of temporal ordering (i.e. is my district moving right or left over time) -- and options like drawing arrows don't fit the small space.

So this is a space both to discuss what to do with the District page to display nominate (use a 1d histogram?) AND to discuss robustifying the decoration setup to be size invariant.

**Comment by JeffreyBLewis** [comment]
Yes, I should make all of the decorations size invariant.   However, I think it would be easier/cooler to make the the red an blue line the that runs through the district table list reflect the 1d nominate score. The rest I think could be found be clicks to the particular member page.

**Comment by aaronrudkin** [comment]
That's... an excellent idea. I will do that.

**Comment by lukesonnet** [comment]
Can we have the median or at least a thin grey 0 line running through? hard to see what it means for districts that are largely one party. Also, that damn incumbency advantage thing hides the coolness of this display. Any way to more clearly signal it? Perhaps a more glamorous column title?

**Comment by aaronrudkin** [comment]
There are a few reasons why it's difficult to layer a second line in this context but I agree that's a good idea. In the mean time, I've added tooltips for the actual score as you mouse over.

When you say incumbency advantage, what do you mean?

**Comment by lukesonnet** [comment]
It wasn't important. The solution you've implemented with the boundaries works very nicely.

**Comment by aaronrudkin** [comment]
Related to the above issue of porting the NOMINATE plot to be size independent, the tooltip code has magic constants for margin sizes in it. (Not a nag, Jeff, just wanted to write it here so we know for further followup)

**Comment by JeffreyBLewis** [comment]
No doubt.  decorate.js really needs to be re-engineered to rely more conventionally on d3 and which should allow more flexibility.   Ultimately, the problem to solve is figuring out how DC maps/places the scatter of points on the SVG canvas.   Right now that is done by inspection.  If it can be done by logic, we are home free.

**Comment by JeffreyBLewis** [comment]
After at bunch of reengineering (head scratching), the scatterplot is a lot more flexible.  Setting the width and margin in voteCharts.js is all that is needed.  

The symbols, axis text and tick marks do not vary with the size of the plot, so better to probably better to leave the margins alone.   

The tool tip no longer relies on magic constants.  Note that the height of the plot is determined by the aspect ratio of the ellipse which in turn depends up the value of nomDWeight.  For this reason care must be take not rely on the exact height of the plot.  A

ll-in-all a major upgrade.
